nomos-core/chain-defs/src/tx/select.rs

```diff
@@ -11,18 +11,17 @@ pub struct FillSize<const SIZE: usize, Tx> {
 }
 
 impl<const SIZE: usize, Tx> FillSize<SIZE, Tx> {
-    pub fn new() -> Self {
-        Self {
-            _tx: Default::default(),
-        }
+    #[must_use]
+    pub const fn new() -> Self {
+        Self { _tx: PhantomData }
     }
 }
 
 impl<const SIZE: usize, Tx: Transaction> TxSelect for FillSize<SIZE, Tx> {
     type Tx = Tx;
     type Settings = ();
 
-    fn new(_: Self::Settings) -> Self {
+    fn new((): Self::Settings) -> Self {
```

Comment:
Is this so in case `Self::Settings` change type the compiler will throw an error about it for this non-usage?

Reply:
Yes. The lint is `clippy::ignored_unit_patterns`.

Reply:
And also because we would notice the change in type itself, not just because of the missing underscore.